### DISCUSSION TITLE
Is there a way to install winget silently/unattended

### DISCUSSION THREAD

**Discussion by Houseton** [microsoft/winget-cli#3925] [Q&A]
I'm looking for a way to install winget application into a newly imaged Windows 10 machine without updates. I've used this code from winget.pro
```
# Get the download URL of the latest winget installer from GitHub:
$API_URL = "https://api.github.com/repos/microsoft/winget-cli/releases/latest"
$DOWNLOAD_URL = $(Invoke-RestMethod $API_URL).assets.browser_download_url |
    Where-Object {$_.EndsWith(".msixbundle")}

# Download the installer:
Invoke-WebRequest -URI $DOWNLOAD_URL -OutFile winget.msixbundle -UseBasicParsing

# Install winget:
Add-AppxPackage winget.msixbundle

# Remove the installer:
Remove-Item winget.msixbundle
```

But it still asks to accept the source agreement for the msstore. I'm trying to incorporate it into a script that runs after imaging and would like it to not need any input. I tried --accept-source-agreements but that is only used in winget itself and not Add-AppxPackage

**Comment by mdanish-kh** [comment]
> I tried --accept-source-agreements but that is only used in winget itself and not Add-AppxPackage

I've not seen Add-AppxPackage ever asking for source agreements 👀 

Can you try the [Sandbox Script at Microsoft Learn](https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox) and see if that does anything differently for you?

**Comment by Trenly** [comment]
If you’re using winget to install a package, you will likely need both `--accept-source-agreements` and `--accept-package-agreements` since some packages have their own specific set of agreements in addition to the source agreements

**Reply by Houseton** [reply]
Thank you for the tip.

**Reply by RottenRonnie** [reply]
I'd be interested in seeing your revised script for this. Thank you

**Reply by Houseton** [reply]
> I'd be interested in seeing your revised script for this. Thank you

If you are talking about my script, I abandoned the problem as I couldn't get it to work and was installing manually. Eventually we moved to an MDM solution and use that now once connected to push updates and packages to the machine.

**Comment by koppor** [comment]
Once could check out https://learn.microsoft.com/en-us/windows/package-manager/winget/

```powershell
$progressPreference = 'silentlyContinue'
Write-Information "Downloading WinGet and its dependencies..."
Invoke-WebRequest -Uri https://aka.ms/getwinget -OutFile Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
Invoke-WebRequest -Uri https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx -OutFile Microsoft.VCLibs.x64.14.00.Desktop.appx
Invoke-WebRequest -Uri https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.8.6/Microsoft.UI.Xaml.2.8.x64.appx -OutFile Microsoft.UI.Xaml.2.8.x64.appx
Add-AppxPackage Microsoft.VCLibs.x64.14.00.Desktop.appx
Add-AppxPackage Microsoft.UI.Xaml.2.8.x64.appx
Add-AppxPackage Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle
```

Tested it on Windows Sandbox with an "Administrator: PowerShell"

**Reply by koppor** [reply]
Update: Did not work for Vagrant here. I found https://github.com/microsoft/winget-cli/issues/3122, also discussing it.

**Comment by koppor** [comment]
It seems that following does the trick

```powershell
    Install-Module -Name WingetTools -Repository PSGallery -Force
    Install-WinGet

    winget install -e --id Mozilla.Firefox --accept-source-agreements
```

**Reply by koppor** [reply]
Update: `WingetTools` is not by Microsoft- Using the official tooling, the command are:

```powershell
    # Source: https://github.com/microsoft/winget-cli/discussions/4065#discussioncomment-8355261
    Install-PackageProvider -Name NuGet -Force
    # Microsoft.WinGet.Client is the official Microsoft winget PowerShell distribution. "WingetTools " is not.
    Install-Module -Name Microsoft.WinGet.Client -Force -Repository PSGallery
    Repair-WinGetPackageManager
```

**Reply by danpetitt** [reply]
When I run this in unattended mode I get a "Error: Specified method is not supported" on the `Repair-WinGetPackageManager` command; but when I log onto the server and run it, its all good.

Why does it not work as expected?

**Reply by tjmoore** [reply]
Maybe `-AllUsers` is required unattended?

```pwsh
Repair-WinGetPackageManager -AllUsers
```

As documented here for Windows Sandbox at least, https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox

```pwsh
$progressPreference = 'silentlyContinue'
Write-Host "Installing WinGet PowerShell module from PSGallery..."
Install-PackageProvider -Name NuGet -Force | Out-Null
Install-Module -Name Microsoft.WinGet.Client -Force -Repository PSGallery | Out-Null
Write-Host "Using Repair-WinGetPackageManager cmdlet to bootstrap WinGet..."
Repair-WinGetPackageManager -AllUsers
Write-Host "Done."
```

**Reply by tjmoore** [reply]
Though in Windows Sandbox I get "Repair-WinGetPackageManager : One or more errors occurred"

However, winget is then installed.
